user_management.c

```diff
@@ -0,0 +1,129 @@
+    fclose(file);
+    printf("User added\n");
+}
+
+void read_users() {
+    User user;
+    FILE* file = fopen("users.txt", "r");
+    if (!file) {
+        printf("Error\n");
+        return;
+    }
+    printf("| ID | Name                 | Age |\n");
+    while (fscanf(file, "%d %49s %d", &user.id, user.name, &user.age) != EOF) {
+        printf("| %2d | %-20s | %3d |\n", user.id, user.name, user.age);
+    }
+    fclose(file);
+}
+
+void update_user() {
+    FILE* file = fopen("users.txt", "r");
+    if (!file) {
+        printf("Error opening file\n");
+        return;
+    }
+    FILE* temp = fopen("temp.txt", "w");
+    if (!temp) {
+        fclose(file);
+        printf("Error creating file\n");
+        return;
+    }
+    int id, found = 0;
+    User user;
+    printf("Enter ID to update: ");
+    scanf("%d", &id);
+    while (fscanf(file, "%d %49s %d", &user.id, user.name, &user.age) != EOF) {
+        if (user.id == id) {
+            printf("Enter new ID: ");
+            scanf("%d", &user.id);
+            getchar(); 
+            printf("Enter new Name: ");
+            fgets(user.name, sizeof(user.name), stdin);
+            user.name[strcspn(user.name, "\n")] = '\0'; 
+            printf("Enter new Age: ");
+            scanf("%d", &user.age);
+            found = 1;
+        }
+        fprintf(temp, "%d %s %d\n", user.id, user.name, user.age);
+    }
+    fclose(file);
+    fclose(temp);
+    remove("users.txt");
+    rename("temp.txt", "users.txt");
+    if (!found) printf("User not found\n");
+    else printf("User updated successfully\n");
+}
+
+void delete_user() {
+    FILE* file = fopen("users.txt", "r");
+    if (!file) {
+        printf("Error opening file\n");
+        return;
+    }
+    FILE* temp = fopen("temp.txt", "w");
+    if (!temp) {
+        fclose(file);
+        printf("Error creating file\n");
+        return;
+    }
+    int id, found = 0;
+    User user;
+    printf("Enter ID : ");
+    scanf("%d", &id);
+    while (fscanf(file, "%d %49s %d", &user.id, user.name, &user.age) != EOF) {
+        if (user.id != id) {
+            fprintf(temp, "%d %s %d\n", user.id, user.name, user.age);
+        } else {
+            found = 1;
+        }
+    }
+    fclose(file);
+    fclose(temp);
+    remove("users.txt");
+    rename("temp.txt", "users.txt");
+    if (!found) printf("User not found\n");
+    else printf("User deleted successfully\n");
+}
+
+int main() {
+    int choice;
+    while (1) {
+        printf("\n1. Create User\n2. Read Users\n3. Update User\n4. Delete User\n5. Exit\nEnter your choice: ");
+        scanf("%d", &choice);
+        getchar(); 
+        switch (choice) {
+            case 1: create_user(); break;
+            case 2: read_users(); break;
+            case 3: update_user(); break;
+            case 4: delete_user(); break;
+            case 5: printf("Exiting...\n"); return 0;
+            default: printf("Invalid choice\n");
```

Comment:
You should also add a functionality that if I enter invalid choice more than let's say 3 time then it should exit